core/templates/services/playthrough-issues-backend-api.service.ts

```diff
@@ -15,87 +15,117 @@
 /**
  * @fileoverview Service for fetching issues and playthroughs from the backend.
  */
-require('domain/statistics/PlaythroughObjectFactory.ts');
-require('domain/statistics/PlaythroughIssueObjectFactory.ts');
-require('domain/utilities/url-interpolation.service.ts');
 
-require('services/services.constants.ajs.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { throwError } from 'rxjs';
 
-angular.module('oppia').factory('PlaythroughIssuesBackendApiService', [
-  '$http', '$q', 'PlaythroughIssueObjectFactory', 'PlaythroughObjectFactory',
-  'UrlInterpolationService', 'FETCH_ISSUES_URL', 'FETCH_PLAYTHROUGH_URL',
-  'RESOLVE_ISSUE_URL',
-  function(
-      $http, $q, PlaythroughIssueObjectFactory, PlaythroughObjectFactory,
-      UrlInterpolationService, FETCH_ISSUES_URL, FETCH_PLAYTHROUGH_URL,
-      RESOLVE_ISSUE_URL) {
-    /** @type {PlaythroughIssue[]} */
-    var cachedIssues = null;
+import { PlaythroughIssueObjectFactory, PlaythroughIssue } from
+  'domain/statistics/PlaythroughIssueObjectFactory.ts';
+import { ServicesConstants } from 'services/services.constants.ts';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service.ts';
 
-    var getFullIssuesUrl = function(explorationId) {
-      return UrlInterpolationService.interpolateUrl(
-        FETCH_ISSUES_URL, {
-          exploration_id: explorationId
-        });
-    };
+@Injectable({
+  providedIn: 'root'
+})
+export class PlaythroughIssuesBackendApiService {
+  constructor(
+    private httpClient: HttpClient,
+    private playthroughIssueObjectFactory: PlaythroughIssueObjectFactory,
+    private urlInterpolationService: UrlInterpolationService) {}
 
-    var getFullPlaythroughUrl = function(expId, playthroughId) {
-      return UrlInterpolationService.interpolateUrl(
-        FETCH_PLAYTHROUGH_URL, {
-          exploration_id: expId,
-          playthrough_id: playthroughId
-        });
-    };
+  private cachedIssues = null;
+
+  private getFullIssuesUrl(explorationId: string): string {
```

Comment:
Please move all private functions to the end of the class, and public functions to the beginning.